transport/src/main/java/io/camunda/zeebe/transport/stream/impl/ClientStreamRequestManager.java

```diff
@@ -313,29 +311,47 @@ private void sendRemoveRequest(
             REQUEST_TIMEOUT);
     registration.setPendingRequest(pendingRequest);
     pendingRequest.whenCompleteAsync(
-        (ok, error) -> handleRemoveResponse(registration, request, error), executor::run);
+        (response, error) -> handleRemoveResponse(registration, request, response, error),
+        executor::run);
   }
 
   private void handleRemoveResponse(
-      final ClientStreamRegistration<M> registration, final byte[] request, final Throwable error) {
+      final ClientStreamRegistration<M> registration,
+      final byte[] request,
+      final byte[] responseBuffer,
+      final Throwable error) {
     final var state = registration.state();
     if (state != State.REMOVING) {
       LOGGER.trace("Skip handling REMOVE response since the state is {}", state, error);
       return;
     }
 
+    final Throwable failure;
+    final Either<ErrorResponse, RemoveStreamResponse> response;
     if (error == null) {
-      registration.transitionToRemoved();
-      purgeRegistration(registration.streamId(), registration.serverId());
-      return;
+      response = responseDecoder.decode(responseBuffer, new RemoveStreamResponse());
+      if (response.isRight()) {
+        registration.transitionToRemoved();
+        purgeRegistration(registration.streamId(), registration.serverId());
+        return;
+      }
+
+      failure = new StreamResponseException(response.getLeft());
+    } else {
+      failure = error;
     }
 
-    // TODO: use backoff delay
```

Comment:
❓ this todo is still valid or?

Reply:
It's already using back off. It's fixed back off, but it's there - we don't retry immediately. AFAIK it was already outdated and we simply forgot it. Now if we want to use exponential back off or something that's a different thing :smile:

Reply:
Ah ok I understood you want to have more the exponential thingy

Reply:
Hm, that wasn't the original todo, but...maybe we do? I'm thinking, if I have hundreds of streams (which isn't that crazy considering they are each fairly low cost), and we restart all the requests at the same time, it's a big spike. The exponential with jitter utility would help spread them out here. It's hard to say if this will really be an issue though since the requests themselves are fairly lightweight :thinking: I suppose spikes would result in time outs, which would result in more requests :thinking: 

Maybe we should add it.

Reply:
https://github.com/camunda/zeebe/issues/15385